sandbox-backend/src/main/java/com/exadel/sandbox/team5/service/impl/MailSenderServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.exadel.sandbox.team5.service.impl;
+
+import com.exadel.sandbox.team5.dao.EmployeeDAO;
+import com.exadel.sandbox.team5.entity.Category;
+import com.exadel.sandbox.team5.entity.Discount;
+import com.exadel.sandbox.team5.service.MailSenderService;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.stereotype.Service;
+import freemarker.template.Configuration;
+import org.springframework.ui.freemarker.FreeMarkerTemplateUtils;
+
+import javax.mail.MessagingException;
+import javax.transaction.Transactional;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Transactional
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class MailSenderServiceImpl implements MailSenderService {
+
+    private final JavaMailSender sender;
+    private final EmployeeDAO employeeDAO;
+    private final Configuration freemarker;
+
+    @Override
+    public void sendEmailToUsers(String notification) {
+        var message = sender.createMimeMessage();
+        var helper = new MimeMessageHelper(message);
+        freemarker.setClassForTemplateLoading(this.getClass(), "/template");
```

Comment:
Please set it during bean configuration.

Reply:
done